src/selectors/path/PathSelector.js

```diff
@@ -162,22 +162,34 @@ class PathSelector extends Selector {
 		 * @type {Sequence}
 		 */
 		const result = this._stepSelectors.reduce(function (intermediateResultNodesSequence, selector) {
-			/**
-			 * @type {Iterator<../DynamicContext>}
-			 */
-			const childContextIterator = dynamicContext.createSequenceIterator(intermediateResultNodesSequence);
-
+			let childContextIterator;
+			if (intermediateResultNodesSequence === null) {
+				// first call, we should use the current dynamic context
+				let hasPassedFirst = false;
+				childContextIterator = {
+					next: () => {
+						if (hasPassedFirst) {
+							return { done: true };
+						}
+						hasPassedFirst = true;
+						return { done: false, value: dynamicContext };
+					}
+				};
+			}
+			else {
+				childContextIterator = dynamicContext.createSequenceIterator(intermediateResultNodesSequence);
+			}
 			/**
 			 * @type {!IteratorIterable<!Sequence>}
 			 */
-			let resultValuesInOrderOfEvaluation = /** @type {!IteratorIterable<!../dataTypes/Sequence>} */ ({
+			let resultValuesInOrderOfEvaluation = /** @type {!IteratorIterable<!Sequence>} */ ({
 				[Symbol.iterator]: () => resultValuesInOrderOfEvaluation,
 				next: () => {
 					const childContext = childContextIterator.next();
 					if (childContext.done) {
 						return { done: true };
 					}
-					if (!isInstanceOfType(childContext.value.contextItem, 'node()')) {
+					if (childContext.value.contextItem !== null && !isInstanceOfType(childContext.value.contextItem, 'node()')) {
```

Comment:
I'd say that should be an error, absent context is different from wrong context.